--- zapret-71.orig/nfq/nfqws.c
+++ zapret-71/nfq/nfqws.c
@@ -223,15 +223,29 @@
 
 	DLOG_CONDUP("unbinding existing nf_queue handler for AF_INET (if any)\n");
 	if (nfq_unbind_pf(*h, AF_INET) < 0) {
-		DLOG_PERROR("nfq_unbind_pf()");
+		DLOG_PERROR("nfq_unbind_pf() AF_INET");
 		goto exiterr;
 	}
 
 	DLOG_CONDUP("binding nfnetlink_queue as nf_queue handler for AF_INET\n");
 	if (nfq_bind_pf(*h, AF_INET) < 0) {
-		DLOG_PERROR("nfq_bind_pf()");
+		DLOG_PERROR("nfq_bind_pf() AF_INET");
 		goto exiterr;
 	}
+	else
+		DLOG_CONDUP("binding for AF_INET success\n");
+
+#ifdef USE_IPV6
+	DLOG_CONDUP("unbinding existing nf_queue handler for AF_INET6 (if any)\n");
+	if (nfq_unbind_pf(*h, AF_INET6) < 0)
+		DLOG_PERROR("nfq_unbind_pf() AF_INET6");
+	
+	DLOG_CONDUP("binding nfnetlink_queue as nf_queue handler for AF_INET6\n");
+	if (nfq_bind_pf(*h, AF_INET6) < 0)
+		DLOG_PERROR("nfq_bind_pf() AF_INET6");
+	else
+		DLOG_CONDUP("binding for AF_INET6 success\n");
+#endif
 
 	DLOG_CONDUP("binding this socket to queue '%u'\n", params.qnum);
 	*qh = nfq_create_queue(*h, params.qnum, &nfq_cb, &params);
