diff -urw flac-1.5.0-orig/Makefile.am flac-1.5.0/Makefile.am
--- flac-1.5.0-orig/Makefile.am	2025-02-10 16:20:26.470898000 +0200
+++ flac-1.5.0/Makefile.am	2025-04-01 02:16:06.914029401 +0300
@@ -31,7 +31,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = doc include m4 man src test oss-fuzz
+SUBDIRS = include m4 src
 
 if EXAMPLES
 SUBDIRS += examples
diff -urw flac-1.5.0-orig/src/Makefile.am flac-1.5.0/src/Makefile.am
--- flac-1.5.0-orig/src/Makefile.am	2025-02-10 16:20:26.478898000 +0200
+++ flac-1.5.0/src/Makefile.am	2025-04-01 02:16:06.915029369 +0300
@@ -27,13 +27,7 @@
 SUBDIRS = \
 	libFLAC \
 	share \
 	$(PROGRAMS_DIRS) \
-	test_grabbag \
-	test_libs_common \
-	test_libFLAC \
-	test_seeking \
-	test_streams \
-	utils \
 	$(CPPLIBS_DIRS)
 
 EXTRA_DIST = \
