diff -urNw wpa_supplicant-2.9-ref/Makefile wpa_supplicant-2.9/Makefile
--- wpa_supplicant-2.9-ref/Makefile	1970-01-01 03:00:00.000000000 +0300
+++ wpa_supplicant-2.9/Makefile	2024-10-04 22:20:41.725895549 +0300
@@ -0,0 +1,23 @@
+SRC_NAME=wpa_supplicant
+
+
+all:
+	cp -f $(SRC_NAME)/.config.def $(SRC_NAME)/.config
+ifeq ($(CONFIG_FIRMWARE_INCLUDE_EAP_PEAP),y)
+	echo CONFIG_EAP_TTLS=y >> $(SRC_NAME)/.config
+	echo CONFIG_EAP_PEAP=y >> $(SRC_NAME)/.config
+	echo CONFIG_EAP_MSCHAPV2=y >> $(SRC_NAME)/.config
+else ifeq (,$(wildcard $(STAGEDIR)/lib/pkgconfig/openssl.pc))
+	echo CONFIG_TLS=internal >> $(SRC_NAME)/.config
+	echo CONFIG_CRYPTO=internal >> $(SRC_NAME)/.config
+	echo CONFIG_INTERNAL_LIBTOMMATH=y >> $(SRC_NAME)/.config
+	echo CONFIG_INTERNAL_RC4=y >> $(SRC_NAME)/.config
+endif
+	$(MAKE) -j$(HOST_NCPU) -C $(SRC_NAME)
+
+clean:
+	$(MAKE) -C $(SRC_NAME) clean
+
+romfs:
+	$(ROMFSINST) $(SRC_NAME)/wpa_cli /usr/sbin/wpa_cli
+	$(ROMFSINST) $(SRC_NAME)/wpa_supplicant /usr/sbin/wpa_supplicant
diff -urNw wpa_supplicant-2.9-ref/wpa_supplicant/.config.def wpa_supplicant-2.9/wpa_supplicant/.config.def
--- wpa_supplicant-2.9-ref/wpa_supplicant/.config.def	1970-01-01 03:00:00.000000000 +0300
+++ wpa_supplicant-2.9/wpa_supplicant/.config.def	2024-10-04 22:20:41.726895474 +0300
@@ -0,0 +1,17 @@
+CONFIG_DRIVER_WIRED=y
+
+CONFIG_IEEE8021X_EAPOL=y
+CONFIG_EAP_MD5=y
+CONFIG_NO_WPA=y
+CONFIG_NO_WPA2=y
+CONFIG_NO_WPA_PASSPHRASE=y
+CONFIG_NO_AES_EXTRAS=y
+CONFIG_BACKEND=file
+
+CONFIG_CTRL_IFACE=unix
+
+CONFIG_NO_STDOUT_DEBUG=y
+CONFIG_NO_SCAN_PROCESSING=y
+CONFIG_NO_CONFIG_WRITE=y
+CONFIG_NO_CONFIG_BLOBS=y
+
diff -urN wpa_supplicant-2.9-orig/wpa_supplicant/Makefile wpa_supplicant-2.9/wpa_supplicant/Makefile
--- wpa_supplicant-2.9-orig/wpa_supplicant/Makefile	2019-08-07 16:25:25.000000000 +0300
+++ wpa_supplicant-2.9/wpa_supplicant/Makefile	2024-10-04 22:20:41.726895474 +0300
@@ -25,5 +25,7 @@
 CFLAGS += $(EXTRA_CFLAGS)
 CFLAGS += -I$(abspath ../src)
 CFLAGS += -I$(abspath ../src/utils)
+CFLAGS += -ffunction-sections -fdata-sections -fvisibility=hidden
+LDFLAGS += -Wl,--gc-sections
 
 -include .config
@@ -887,6 +889,7 @@
 CFLAGS += -DIEEE8021X_EAPOL
 OBJS += ../src/eapol_supp/eapol_supp_sm.o
 OBJS += ../src/eap_peer/eap.o ../src/eap_peer/eap_methods.o
+NEED_RC4=y
 NEED_EAP_COMMON=y
