diff -urN curl-8.0.1-orig/Makefile.am curl-8.0.1/Makefile.am
--- curl-8.0.1-orig/Makefile.am	2023-03-18 01:34:19.000000000 +0200
+++ curl-8.0.1/Makefile.am	2023-03-20 23:33:14.435430034 +0200
@@ -158,7 +158,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+DIST_SUBDIRS = $(SUBDIRS) packages scripts include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -272,8 +272,6 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
diff -urN curl-8.0.1-orig/configure.ac curl-8.0.1/configure.ac
--- curl-8.0.1-orig/configure.ac	2023-03-20 13:38:42.000000000 +0200
+++ curl-8.0.1/configure.ac	2023-03-20 23:33:14.439429023 +0200
@@ -4648,29 +4648,14 @@
 fi
 
 AC_CONFIG_FILES([Makefile \
-           docs/Makefile \
-           docs/examples/Makefile \
-           docs/libcurl/Makefile \
-           docs/libcurl/opts/Makefile \
-           docs/cmdline-opts/Makefile \
            include/Makefile \
            include/curl/Makefile \
            src/Makefile \
            lib/Makefile \
            scripts/Makefile \
            lib/libcurl.vers \
            lib/libcurl.plist \
-           tests/Makefile \
-           tests/config \
-           tests/certs/Makefile \
-           tests/certs/scripts/Makefile \
-           tests/data/Makefile \
-           tests/server/Makefile \
-           tests/libtest/Makefile \
-           tests/unit/Makefile \
-           tests/http/config.ini \
-           tests/http/Makefile \
            packages/Makefile \
            packages/vms/Makefile \
            curl-config \
            libcurl.pc
diff -urN curl-8.0.1-orig/src/Makefile.am curl-8.0.1/src/Makefile.am
--- curl-8.0.1-orig/src/Makefile.am	2023-03-18 01:34:19.000000000 +0200
+++ curl-8.0.1/src/Makefile.am	2023-03-20 23:33:14.439429023 +0200
@@ -45,7 +45,7 @@
 
 bin_PROGRAMS = curl
 
-SUBDIRS = ../docs
+SUBDIRS =
 
 if USE_CPPFLAG_CURL_STATICLIB
 AM_CPPFLAGS += -DCURL_STATICLIB
