diff -urN curl-7.77.0-ref/Makefile.am curl-7.77.0/Makefile.am
--- curl-7.77.0-ref/Makefile.am	2021-04-12 10:31:13.000000000 +0300
+++ curl-7.77.0/Makefile.am	2021-07-08 14:08:40.951714100 +0300
@@ -195,7 +195,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+DIST_SUBDIRS = $(SUBDIRS) packages scripts include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -306,8 +306,6 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
diff -urN curl-7.77.0-ref/configure.ac curl-7.77.0/configure.ac
--- curl-7.77.0-ref/configure.ac	2021-05-25 13:50:36.000000000 +0300
+++ curl-7.77.0/configure.ac	2021-07-08 14:09:15.204592800 +0300
@@ -173,7 +173,7 @@
      curl_h3_msg="no      (--with-ngtcp2, --with-quiche)"
 
 enable_altsvc="yes"
-enable_hsts="yes"
+enable_hsts="no"
 
 dnl
 dnl Save some initial values the user might have provided
@@ -3935,7 +3935,7 @@
 
 if test "x$enable_hsts" != "xyes"; then
   curl_hsts_msg="no      (--enable-hsts)";
-  AC_DEFINE(CURL_DISABLE_HSTS, 1, [disable alt-svc])
+  AC_DEFINE(CURL_DISABLE_HSTS, 1, [disable HSTS])
 fi
 
 dnl *************************************************************
@@ -4254,24 +4254,12 @@
 fi
 
 AC_CONFIG_FILES([Makefile \
-           docs/Makefile \
-           docs/examples/Makefile \
-           docs/libcurl/Makefile \
-           docs/libcurl/opts/Makefile \
-           docs/cmdline-opts/Makefile \
            include/Makefile \
            include/curl/Makefile \
            src/Makefile \
            lib/Makefile \
            scripts/Makefile \
            lib/libcurl.vers \
-           tests/Makefile \
-           tests/certs/Makefile \
-           tests/certs/scripts/Makefile \
-           tests/data/Makefile \
-           tests/server/Makefile \
-           tests/libtest/Makefile \
-           tests/unit/Makefile \
            packages/Makefile \
            packages/vms/Makefile \
            curl-config \
@@ -4281,7 +4269,7 @@
 
 CURL_GENERATE_CONFIGUREHELP_PM
 
-XC_AMEND_DISTCLEAN([lib src tests/unit tests/server tests/libtest docs/examples])
+XC_AMEND_DISTCLEAN([lib src])
 
 AC_MSG_NOTICE([Configured to build curl/libcurl:
 
diff -urN curl-7.77.0-ref/src/Makefile.am curl-7.77.0/src/Makefile.am
--- curl-7.77.0-ref/src/Makefile.am	2021-04-12 10:31:13.000000000 +0300
+++ curl-7.77.0/src/Makefile.am	2021-07-08 14:08:40.951714100 +0300
@@ -43,7 +43,7 @@
 
 bin_PROGRAMS = curl
 
-SUBDIRS = ../docs
+SUBDIRS =
 
 if USE_CPPFLAG_CURL_STATICLIB
 AM_CPPFLAGS += -DCURL_STATICLIB
