diff -urN curl-7.88.0-orig/Makefile.am curl-7.88.0/Makefile.am
--- curl-7.88.0-orig/Makefile.am	2023-02-13 09:37:04.000000000 +0200
+++ curl-7.88.0/Makefile.am	2023-02-20 03:52:42.616759187 +0200
@@ -158,7 +158,7 @@
 bin_SCRIPTS = curl-config
 
 SUBDIRS = lib src
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+DIST_SUBDIRS = $(SUBDIRS) packages scripts include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -272,8 +272,6 @@
 # We extend the standard install with a custom hook:
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
diff -urN curl-7.88.0-orig/configure.ac curl-7.88.0/configure.ac
--- curl-7.88.0-orig/configure.ac	2023-02-13 09:37:04.000000000 +0200
+++ curl-7.88.0/configure.ac	2023-02-20 03:52:42.616759187 +0200
@@ -4643,29 +4643,14 @@
 fi
 
 AC_CONFIG_FILES([Makefile \
-           docs/Makefile \
-           docs/examples/Makefile \
-           docs/libcurl/Makefile \
-           docs/libcurl/opts/Makefile \
-           docs/cmdline-opts/Makefile \
            include/Makefile \
            include/curl/Makefile \
            src/Makefile \
            lib/Makefile \
            scripts/Makefile \
            lib/libcurl.vers \
            lib/libcurl.plist \
-           tests/Makefile \
-           tests/config \
-           tests/certs/Makefile \
-           tests/certs/scripts/Makefile \
-           tests/data/Makefile \
-           tests/server/Makefile \
-           tests/libtest/Makefile \
-           tests/unit/Makefile \
-           tests/tests-httpd/config.ini \
-           tests/tests-httpd/Makefile \
            packages/Makefile \
            packages/vms/Makefile \
            curl-config \
            libcurl.pc
@@ -4674,7 +4659,7 @@
 
 CURL_GENERATE_CONFIGUREHELP_PM
 
-XC_AMEND_DISTCLEAN([lib src tests/unit tests/server tests/libtest docs/examples])
+XC_AMEND_DISTCLEAN([lib src])
 
 AC_MSG_NOTICE([Configured to build curl/libcurl:
 
diff -urN curl-7.88.0-orig/src/Makefile.am curl-7.88.0/src/Makefile.am
--- curl-7.88.0-orig/src/Makefile.am	2023-02-13 09:37:04.000000000 +0200
+++ curl-7.88.0/src/Makefile.am	2023-02-20 03:52:42.616759187 +0200
@@ -45,7 +45,7 @@
 
 bin_PROGRAMS = curl
 
-SUBDIRS = ../docs
+SUBDIRS =
 
 if USE_CPPFLAG_CURL_STATICLIB
 AM_CPPFLAGS += -DCURL_STATICLIB
