diff -urN curl-8.14.1-orig/Makefile.am curl-8.14.1/Makefile.am
--- curl-8.14.1-orig/Makefile.am	2025-06-04 08:42:09.000000000 +0300
+++ curl-8.14.1/Makefile.am	2025-06-22 18:30:39.244684467 +0300
@@ -92,7 +92,7 @@
 bin_SCRIPTS = curl-config
 
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests packages include docs
+SUBDIRS = lib src
+DIST_SUBDIRS = $(SUBDIRS) packages include
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
@@ -187,14 +187,11 @@
 # We extend the standard install with a custom hook:
 if BUILD_DOCS
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
-	(cd docs/libcurl && $(MAKE) install)
 else
 install-data-hook:
 	(cd include && $(MAKE) install)
-	(cd docs && $(MAKE) install)
 endif
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
diff -rupw curl-8.14.1-orig/Makefile.in curl-8.14.1/Makefile.in
--- curl-8.14.1-orig/Makefile.in	2025-06-04 08:42:09.000000000 +0300
+++ curl-8.14.1/Makefile.in	2025-06-22 18:30:39.245684364 +0300
@@ -508,7 +508,7 @@
 
 bin_SCRIPTS = curl-config
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests packages include docs
+SUBDIRS = lib src
+DIST_SUBDIRS = $(SUBDIRS) packages include
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 LIB_VAUTH_CFILES = \
@@ -1143,11 +1143,8 @@
 # We extend the standard install with a custom hook:
 @BUILD_DOCS_TRUE@install-data-hook:
 @BUILD_DOCS_TRUE@	(cd include && $(MAKE) install)
-@BUILD_DOCS_TRUE@	(cd docs && $(MAKE) install)
-@BUILD_DOCS_TRUE@	(cd docs/libcurl && $(MAKE) install)
 @BUILD_DOCS_FALSE@install-data-hook:
 @BUILD_DOCS_FALSE@	(cd include && $(MAKE) install)
-@BUILD_DOCS_FALSE@	(cd docs && $(MAKE) install)
 
 # We extend the standard uninstall with a custom hook:
 uninstall-hook:
diff -urN curl-8.14.1-orig/configure.ac curl-8.14.1/configure.ac
--- curl-8.14.1-orig/configure.ac	2025-06-04 08:42:09.000000000 +0300
+++ curl-8.14.1/configure.ac	2025-06-22 18:30:39.245684364 +0300
@@ -5471,33 +5471,16 @@
 CURL_PREPARE_CONFIGUREHELP_PM
 
 AC_CONFIG_FILES([\
   Makefile \
-  docs/Makefile \
-  docs/examples/Makefile \
-  docs/libcurl/Makefile \
-  docs/libcurl/opts/Makefile \
-  docs/cmdline-opts/Makefile \
   include/Makefile \
   include/curl/Makefile \
   src/Makefile \
   lib/Makefile \
   scripts/Makefile \
   lib/libcurl.vers \
-  tests/Makefile \
-  tests/config \
-  tests/configurehelp.pm \
-  tests/certs/Makefile \
-  tests/data/Makefile \
-  tests/server/Makefile \
-  tests/libtest/Makefile \
-  tests/unit/Makefile \
-  tests/tunit/Makefile \
-  tests/http/config.ini \
-  tests/http/Makefile \
-  tests/http/clients/Makefile \
   packages/Makefile \
   packages/vms/Makefile \
   curl-config \
   libcurl.pc
 ])
 AC_OUTPUT
