diff -urN curl-8.7.1-orig/Makefile.am curl-8.7.1/Makefile.am
--- curl-8.7.1-orig/Makefile.am	2024-03-26 09:42:20.000000000 +0200
+++ curl-8.7.1/Makefile.am	2024-04-03 01:09:30.970026401 +0300
@@ -135,7 +135,7 @@
 bin_SCRIPTS = curl-config
 
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+SUBDIRS = lib docs src
+DIST_SUBDIRS = $(SUBDIRS) packages include docs
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
diff -rupw curl-8.7.1-orig/Makefile.in curl-8.7.1/Makefile.in
--- curl-8.7.1-orig/Makefile.in	2024-03-27 10:03:42.000000000 +0200
+++ curl-8.7.1/Makefile.in	2024-04-03 01:09:30.970026401 +0300
@@ -666,7 +666,7 @@
 
 bin_SCRIPTS = curl-config
-SUBDIRS = lib docs src scripts
-DIST_SUBDIRS = $(SUBDIRS) tests packages scripts include docs
+SUBDIRS = lib docs src
+DIST_SUBDIRS = $(SUBDIRS) packages include docs
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libcurl.pc
 LIB_VAUTH_CFILES = \
diff -urN curl-8.7.1-orig/configure.ac curl-8.7.1/configure.ac
--- curl-8.7.1-orig/configure.ac	2024-03-27 00:05:31.000000000 +0200
+++ curl-8.7.1/configure.ac	2024-04-03 01:09:30.974026215 +0300
@@ -4950,17 +4950,6 @@
            lib/libcurl.vers \
            lib/libcurl.plist \
-           tests/Makefile \
-           tests/config \
-           tests/certs/Makefile \
-           tests/certs/scripts/Makefile \
-           tests/data/Makefile \
-           tests/server/Makefile \
-           tests/libtest/Makefile \
-           tests/unit/Makefile \
-           tests/http/config.ini \
-           tests/http/Makefile \
-           tests/http/clients/Makefile \
            packages/Makefile \
            packages/vms/Makefile \
            curl-config \
            libcurl.pc
