diff -urN libusb-1.0.28-orig/configure.ac libusb-1.0.28-orig/configure.ac
--- libusb-1.0.28-orig/configure.ac	2025-03-19 00:13:45.000000000 +0200
+++ libusb-1.0.28/configure.ac	2025-03-30 21:39:46.471443585 +0300
@@ -182,8 +182,8 @@ usb_modeswitch fix by removing libatomic dependency from libusb
 	AC_MSG_RESULT($GCC_ATOMIC_BUILTINS_SUPPORTED)
-	if test "x$GCC_ATOMIC_BUILTINS_SUPPORTED" != xyes; then
-		AC_SEARCH_LIBS([__atomic_fetch_add_4], [atomic])
-	fi
+#	if test "x$GCC_ATOMIC_BUILTINS_SUPPORTED" != xyes; then
+#		AC_SEARCH_LIBS([__atomic_fetch_add_4], [atomic])
+#	fi
 elif test "x$platform" = xwindows; then
 	AC_DEFINE([PLATFORM_WINDOWS], [1], [Define to 1 if compiling for a Windows platform.])
 else
 	AC_MSG_ERROR([Unknown platform])
