diff -urw e2fsprogs-1.47.2-ref/misc/mke2fs.conf.in e2fsprogs-1.47.2/misc/mke2fs.conf.in
--- e2fsprogs-1.47.2-ref/misc/mke2fs.conf.in	2024-05-29 01:34:20.699351940 +0000
+++ e2fsprogs-1.47.2/misc/mke2fs.conf.in	2024-05-29 01:51:46.195465734 +0000
@@ -1,6 +1,6 @@
 [defaults]
-	base_features = sparse_super,large_file,filetype,resize_inode,dir_index,ext_attr
-	default_mntopts = acl,user_xattr
+	base_features = sparse_super,filetype,resize_inode,dir_index,ext_attr
+	default_mntopts = user_xattr
 	enable_periodic_fsck = 0
 	blocksize = 4096
 	inode_size = 256
@@ -11,7 +11,7 @@
 		features = has_journal
 	}
 	ext4 = {
-		features = has_journal,extent,huge_file,flex_bg,metadata_csum,metadata_csum_seed,64bit,dir_nlink,extra_isize,orphan_file
+		features = has_journal,extent,huge_file,flex_bg,64bit,dir_nlink,extra_isize
 	}
 	small = {
 		blocksize = 1024
