diff -ur util-linux-2.41-ref/libblkid/src/partitions/dos.c util-linux-2.41/libblkid/src/partitions/dos.c
--- util-linux-2.41-ref/libblkid/src/partitions/dos.c	2025-02-26 14:27:07.512597000 +0200
+++ util-linux-2.41/libblkid/src/partitions/dos.c	2025-03-31 15:22:46.542630595 +0300
@@ -28,10 +28,7 @@
 } dos_nested[] = {
 	{ MBR_FREEBSD_PARTITION, &bsd_pt_idinfo },
 	{ MBR_NETBSD_PARTITION, &bsd_pt_idinfo },
-	{ MBR_OPENBSD_PARTITION, &bsd_pt_idinfo },
-	{ MBR_UNIXWARE_PARTITION, &unixware_pt_idinfo },
-	{ MBR_SOLARIS_X86_PARTITION, &solaris_x86_pt_idinfo },
-	{ MBR_MINIX_PARTITION, &minix_pt_idinfo }
+	{ MBR_OPENBSD_PARTITION, &bsd_pt_idinfo }
 };
 
 static inline int is_extended(struct dos_partition *p)
diff -ur util-linux-2.41-ref/libblkid/src/partitions/partitions.c util-linux-2.41/libblkid/src/partitions/partitions.c
--- util-linux-2.41-ref/libblkid/src/partitions/partitions.c	2025-02-26 14:27:07.512597000 +0200
+++ util-linux-2.41/libblkid/src/partitions/partitions.c	2025-03-31 15:22:46.544630521 +0300
@@ -123,19 +123,11 @@
  */
 static const struct blkid_idinfo *idinfos[] =
 {
-	&aix_pt_idinfo,
-	&sgi_pt_idinfo,
-	&sun_pt_idinfo,
 	&dos_pt_idinfo,
 	&gpt_pt_idinfo,
 	&pmbr_pt_idinfo,	/* always after GPT */
 	&mac_pt_idinfo,
-	&ultrix_pt_idinfo,
-	&bsd_pt_idinfo,
-	&unixware_pt_idinfo,
-	&solaris_x86_pt_idinfo,
-	&minix_pt_idinfo,
-	&atari_pt_idinfo
+	&bsd_pt_idinfo
 };
 
 /*
diff -ur util-linux-2.41-ref/libblkid/src/superblocks/superblocks.c util-linux-2.41/libblkid/src/superblocks/superblocks.c
--- util-linux-2.41-ref/libblkid/src/superblocks/superblocks.c.orig	2025-02-26 14:27:07.515597000 +0200
+++ util-linux-2.41/libblkid/src/superblocks/superblocks.c	2025-03-31 15:22:46.545630483 +0300
@@ -97,89 +97,20 @@
 	/* First, as access to locked OPAL region triggers IO errors */
 	&luks_opal_idinfo,
 
-	/* RAIDs */
-	&linuxraid_idinfo,
-	&ddfraid_idinfo,
-	&iswraid_idinfo,
-	&lsiraid_idinfo,
-	&viaraid_idinfo,
-	&silraid_idinfo,
-	&nvraid_idinfo,
-	&pdcraid_idinfo,
-	&highpoint45x_idinfo,
-	&highpoint37x_idinfo,
-	&adraid_idinfo,
-	&jmraid_idinfo,
-
-	&bcache_idinfo,
-	&bcachefs_idinfo,
-	&bluestore_idinfo,
-	&drbd_idinfo,
-	&drbdmanage_idinfo,
-	&drbdproxy_datalog_idinfo,
-	&lvm2_idinfo,
-	&lvm1_idinfo,
-	&snapcow_idinfo,
-	&verity_hash_idinfo,
-	&integrity_idinfo,
-	&luks_idinfo,
-	&vmfs_volume_idinfo,
-	&ubi_idinfo,
-	&vdo_idinfo,
-	&stratis_idinfo,
-	&bitlocker_idinfo,
-	&cs_fvault2_idinfo,
-
 	/* Filesystems */
 	&vfat_idinfo,
 	&swsuspend_idinfo,
 	&swap_idinfo,
-	&xfs_idinfo,
-	&xfs_log_idinfo,
 	&exfs_idinfo,
 	&ext4dev_idinfo,
 	&ext4_idinfo,
 	&ext3_idinfo,
 	&ext2_idinfo,
 	&jbd_idinfo,
-	&reiser_idinfo,
-	&reiser4_idinfo,
-	&jfs_idinfo,
-	&udf_idinfo,
-	&iso9660_idinfo,
-	&zfs_idinfo,
 	&hfsplus_idinfo,
 	&hfs_idinfo,
-	&ufs_idinfo,
-	&hpfs_idinfo,
-	&sysv_idinfo,
-        &xenix_idinfo,
 	&ntfs_idinfo,
-	&refs_idinfo,
-	&cramfs_idinfo,
-	&romfs_idinfo,
-	&minix_idinfo,
-	&gfs_idinfo,
-	&gfs2_idinfo,
-	&ocfs_idinfo,
-	&ocfs2_idinfo,
-	&oracleasm_idinfo,
-	&vxfs_idinfo,
-	&squashfs_idinfo,
-	&squashfs3_idinfo,
-	&netware_idinfo,
-	&btrfs_idinfo,
-	&ubifs_idinfo,
-	&bfs_idinfo,
-	&vmfs_fs_idinfo,
-	&befs_idinfo,
-	&nilfs2_idinfo,
 	&exfat_idinfo,
-	&f2fs_idinfo,
-	&mpool_idinfo,
-	&apfs_idinfo,
-	&zonefs_idinfo,
-	&erofs_idinfo,
 };
 
 /*
diff -ur util-linux-2.41-ref/libblkid/src/topology/topology.c util-linux-2.41/libblkid/src/topology/topology.c
--- util-linux-2.41-ref/libblkid/src/topology/topology.c	2025-02-26 14:27:07.516597000 +0200
+++ util-linux-2.41/libblkid/src/topology/topology.c	2025-03-31 15:22:46.546630446 +0300
@@ -79,11 +79,7 @@
 {
 #ifdef __linux__
 	&sysfs_tp_idinfo,
-	&ioctl_tp_idinfo,
-	&md_tp_idinfo,
-	&dm_tp_idinfo,
-	&lvm_tp_idinfo,
-	&evms_tp_idinfo
+	&ioctl_tp_idinfo
 #endif
 };
 
diff -ur util-linux-2.41-ref/sys-utils/setarch.c util-linux-2.41/sys-utils/setarch.c
--- util-linux-2.41-ref/sys-utils/setarch.c	2025-02-26 14:27:07.639596000 +0200
+++ util-linux-2.41/sys-utils/setarch.c	2025-03-31 15:22:46.546630446 +0300
@@ -108,7 +108,6 @@
     X(PER_SUNOS) \
     X(PER_XENIX) \
     X(PER_LINUX32) \
-    X(PER_LINUX32_3GB) \
     X(PER_IRIX32) \
     X(PER_IRIXN32) \
     X(PER_IRIX64) \
